components/permissions/isAdmin.tsx

```diff
@@ -0,0 +1,26 @@
+import { RootState } from "@/store/store";
+import { useEffect } from "react";
+import { useSelector } from "react-redux";
+
+import { useRouter } from "next/router";
+
+export default function isAuth(Component: any) {
```

Comment:
Similar to the comment on the server side, could we also get a string parameter for future implementation of "exec" only pages, or "admin" only pages, or "members" only pages. These roles are just examples and no necessarily the role names we would use. For now we just as pass in the string as "admin", but when we want to use it for other roles we can just pass the role name. 

Preferably the param would be a string array, so we can define this page is accessible to only "execs" and "admins" by passing ["exec", "admin"]

I think it would be best to keep the naming convention clear and consistent, right now the file is named isAdmin and the function isAuth, they should be the same. Otherwise I think the name of this component should be authPage what do you think ?